src/main/java/net/minestom/server/network/ComponentNetworkBufferTypeImpl.java

```diff
@@ -247,7 +256,35 @@ private void writeHoverEvent(@NotNull NetworkBuffer buffer, @NotNull HoverEvent<
 
             buffer.write(BYTE, TAG_COMPOUND);
             writeUtf(buffer, "components");
-            //todo item components
+            BinaryTagWriter nbtWriter = impl(buffer).nbtWriter;
```

Comment:
this is duplicated from NbtTypeImpl, this logic should just be moved into NetworkBufferTypeImpl, eg `impl(buffer).nbtWriter()`, same for reader.